src/controllers/scripting/controllerscriptenginebase.cpp

```diff
@@ -194,6 +199,22 @@ void ControllerScriptEngineBase::showQMLExceptionDialog(
         scriptErrorDialog(errorText, errorText, bFatalError);
     }
 }
+
+QObject* ControllerScriptEngineBase::getPlayer(const QString& deck) {
+    if (!s_pPlayerManager) {
+        qWarning() << "PlayerManagerProxy failed to find player for group" << deck;
+        return nullptr;
+    }
+
+    // Don't set a parent here, so that the QML engine deletes the object when
+    // the corresponding JS object is garbage collected.
+    JavascriptPlayerProxy* pPlayerProxy = new JavascriptPlayerProxy(
+            s_pPlayerManager->getPlayer(deck), nullptr);
```

Comment:
`PlayerManager::getPlayer` will return a null ptr if `deck` doesn't exist, could you please handle that gracefully and return a `QJSEngine::newErrorObject`?

Reply:
Happy with the middle ground of returning `null` if we think this is best. I think we shouldn't return an invalid `JavascriptPlayerProxy` tho

Reply:
Yes I changed that in the last commit.

Reply:
Can I resolve this so it's clearer for me what's left?